### PR TITLE
CI: Try to fix codecov upload with circleCI orb

### PR DESCRIPTION
**Pull request by okurz**
Following
https://docs.codecov.com/docs/github-2-getting-a-codecov-account-and-uploading-coverage

Related progress issue: https://progress.opensuse.org/issues/128129

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
-version: 2
+version: 2.1
```
[the remark this answers is not in the document]
Comment (perlpunk) [reply .circleci/config.yml:1]: https://discuss.circleci.com/t/circleci-2-1-config-overview/26057 mentions that there are some changes in 2.1, and no other version fields are allowed anymore.
Maybe we can just stay with version 2 for this.

[the remark this answers is not in the document]
Comment (okurz) [reply .circleci/config.yml:1]: seems like 2.1 is required for the orb syntax, trying to update the config.

**Comment by Martchus** [comment]
I have just allowed running this orb in the security settings. I haven't seen a way to re-trigger the pipeline, though. So you likely need to force-push.

**Comment by perlpunk** [comment]
@Mergifyio rebase

**Comment by perlpunk** [comment]
Unfortunately we get the same error: https://app.circleci.com/pipelines/github/os-autoinst/openQA/11396/workflows/7359cfd5-9581-41f0-b7ed-868283976ce2/jobs/106483?invite=true#step-114-51

**Comment by okurz** [comment]
Trying to specify codecov.json file explicitly plus "extra args". In the latest commit also adding the SHA of the commit with

```
xtra_args: -v -Z -C $CIRCLE_SHA1
```

**Comment by okurz** [comment]
https://app.circleci.com/pipelines/github/os-autoinst/openQA/11416/workflows/dd1b6b97-d19f-4d19-9bb9-d405ee2f6160/jobs/106660 shows the success

**Review by kraih — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**